frontend_server/lib/server.dart

```diff
@@ -0,0 +1,207 @@
+  IncrementalKernelGenerator _ikg;
+  String _filename;
+
+  @override
+  Future<Null> compile(String filename, ArgResults options) async {
+    final String boundaryKey = new Uuid().generateV4();
+    print("result $boundaryKey");
+    final Uri sdkRoot = _ensureFolderPath(options['sdk-root']);
+    final CompilerOptions compilerOptions = new CompilerOptions()
+      ..sdkRoot = sdkRoot
+      ..strongMode = false
+      ..target = new FlutterTarget(new TargetFlags())
+      ..onError = (CompilationMessage message) {
+        print(message);
+      };
+    Program program;
+    if (options['incremental']) {
+      _ikg = await IncrementalKernelGenerator.newInstance(
+          compilerOptions, Uri.base.resolve(filename));
+      final DeltaProgram deltaProgram = await _ikg.computeDelta();
+      program = deltaProgram.newProgram;
+    } else {
+      // TODO(aam): Remove linkedDependencies once platform is directly embedded
+      // into VM snapshot and http://dartbug.com/30111 is fixed.
+      compilerOptions.linkedDependencies = <Uri>[
+        sdkRoot.resolve('platform.dill')
+      ];
+      program = await kernelForProgram(Uri.base.resolve(filename), compilerOptions);
+    }
+    if (program != null) {
+      final String kernelBinaryFilename = filename + ".dill";
+      await writeProgramToBinary(program, kernelBinaryFilename);
+      print(boundaryKey + " " + kernelBinaryFilename);
+    } else {
+      print(boundaryKey);
+    }
+    return null;
+  }
+
+  @override
+  Future<Null> recompileDelta() async {
+    _ikg.computeDelta();
+    final String boundaryKey = new Uuid().generateV4();
+    print("result $boundaryKey");
+    final DeltaProgram deltaProgram = await _ikg.computeDelta();
+    final Program program = deltaProgram.newProgram;
+    final String kernelBinaryFilename = _filename + ".dill";
+    await writeProgramToBinary(program, kernelBinaryFilename);
+    print(boundaryKey);
+    return null;
+  }
+
+  @override
+  void acceptLastDelta() {
+    _ikg.acceptLastDelta();
+  }
+
+  @override
+  void rejectLastDelta() {
+    _ikg.rejectLastDelta();
+  }
+
+  @override
+  void invalidate(Uri uri) {
+    _ikg.invalidate(uri);
+  }
+
+  Uri _ensureFolderPath(String path) {
+    if (!path.endsWith('/')) {
+      path = '$path/';
+    }
+    return Uri.base.resolve(path);
+  }
+}
+
+/// Entry point for this module, that creates `_FrontendCompiler` instance and
+/// processes user input.
+/// `compiler` is an optional parameter so it can be replaced with mocked
+/// version for testing.
+Future<int> starter(List<String> args, {CompilerInterface compiler}) async {
+  final ArgResults options = _argParser.parse(args);
+  if (options['train']) {
+    return 0;
+  }
+
+  compiler ??= new _FrontendCompiler();
+
+  if (options.rest.isNotEmpty) {
+    await compiler.compile(options.rest[0], options);
+    return 0;
+  }
+
+  _State state = _State.READY_FOR_INSTRUCTION;
+  String boundaryKey;
+  print('Frontend server is ready.');
```

Comment:
That string doesn't appear in your docs, so it seems to be a violation of the defined protocol.